pkg/operator/controller/dns/controller.go

```diff
@@ -0,0 +1,178 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+	"github.com/openshift/cluster-ingress-operator/pkg/util/slice"
+
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	runtimecontroller "sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "dns_controller"
+)
+
+var log = logf.Logger.WithName(controllerName)
+
+func New(mgr manager.Manager, operatorNamespace string, dnsProvider dns.Provider) (runtimecontroller.Controller, error) {
+	reconciler := &reconciler{
+		client:            mgr.GetClient(),
+		cache:             mgr.GetCache(),
+		recorder:          mgr.GetEventRecorderFor(controllerName),
+		operatorNamespace: operatorNamespace,
+		dnsProvider:       dnsProvider,
+	}
+	c, err := runtimecontroller.New(controllerName, mgr, runtimecontroller.Options{Reconciler: reconciler})
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &iov1.DNSRecord{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+type reconciler struct {
+	client            client.Client
+	cache             cache.Cache
+	recorder          record.EventRecorder
+	operatorNamespace string
+	dnsProvider       dns.Provider
+	dnsConfig         *configv1.DNS
```

Comment:
Half of these fields are not used:

* `recorder` is not used (but maybe it should be).
* `operatorNamespace` is not used and probably should be deleted.
* `dnsConfig` is not used and should be deleted.

Reply:
Fixed